From 84e6428c540740648d80eac0219630978d5f0c33 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Macholda?= <tomas.macholda@nic.cz>
Date: Mon, 9 Feb 2026 15:11:39 +0100
Subject: [PATCH 2/2] luci-mod-system: support for RGB LEDs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

added support for setting LED brightness and colours

channels and max_brightness are pulled dynamically from sysfs

Signed-off-by: Tomáš Macholda <tomas.macholda@nic.cz>
---
 .../luci-static/resources/view/system/leds.js | 20 +++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/modules/luci-mod-system/htdocs/luci-static/resources/view/system/leds.js b/modules/luci-mod-system/htdocs/luci-static/resources/view/system/leds.js
index f34a9360c3..2bec03cc0e 100644
--- a/modules/luci-mod-system/htdocs/luci-static/resources/view/system/leds.js
+++ b/modules/luci-mod-system/htdocs/luci-static/resources/view/system/leds.js
@@ -107,6 +107,26 @@ return view.extend({
 				var plugin = plugins[i];
 				plugin.form.addFormOptions(s);
 			}
+			var led_names = Object.keys(leds)
+			for (var i = 0; i < led_names.length; i++) {
+				var led = leds[led_names[i]];
+				o = s.option(form.Value, `${led_names[i]}_brightness`, _('Brightness'));
+				o.depends('sysfs', led_names[i])
+				o.ucioption = 'brightness'
+				o.datatype = `range(0,${led.max_brightness})`;
+				o.placeholder = led.max_brightness;
+				o.modalonly = true;
+				if (led.multi_channel) {
+					for (var channel in led.channels) {
+						o = s.option(form.Value, `${led_names[i]}_color_${channel}`, channel);
+						o.ucioption = `color_${channel}`
+						o.depends('sysfs', led_names[i])
+						o.datatype = `range(0,${led.max_brightness})`;
+						o.placeholder = led.channels[channel];
+						o.modalonly = true;
+					}
+				}
+			}
 
 			var opts = s.getOption();
 
-- 
2.53.0

